src/core/cluster.rs

```diff
@@ -90,33 +184,26 @@ impl<'a, T: Number, U: Number, D: Dataset<T, U>> Ord for Cluster<'a, T, U, D> {
 ///
 /// TODO: Add a tree-based prefix to the cluster names when we need to hash
 /// clusters from different trees into the same collection.
-impl<'a, T: Number, U: Number, D: Dataset<T, U>> Hash for Cluster<'a, T, U, D> {
+impl<T: Number, U: Number, D: Dataset<T, U>> Hash for Cluster<T, U, D> {
     fn hash<H: Hasher>(&self, state: &mut H) {
         self.name().hash(state)
     }
 }
 
-impl<'a, T: Number, U: Number, D: Dataset<T, U>> std::fmt::Display for Cluster<'a, T, U, D> {
+impl<T: Number, U: Number, D: Dataset<T, U>> std::fmt::Display for Cluster<T, U, D> {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
         write!(f, "{}", self.name())
     }
 }
 
-impl<'a, T: Number, U: Number, D: Dataset<T, U>> Cluster<'a, T, U, D> {
-    /// Creates a new root `Cluster` for the metric space.
+impl<T: Number, U: Number, D: Dataset<T, U>> Cluster<T, U, D> {/// Creates a new root `Cluster` for the metric space.
     ///
     /// # Arguments
     ///
     /// * `dataset`: on which to create the `Cluster`.
-    pub fn new_root(dataset: &'a D) -> Self {
+    pub fn new_root(data: &D, indices: Vec<usize>, seed: Option<u64>) -> Self {
```

Comment:
Have the `indices` input parameter be of type `&[usize]`. Also see `Cluster::new`